lib/rules-generator.js

```diff
@@ -153,21 +162,28 @@ util.methods(Generator, {
     var thisVar = ast.variable('this');
 
     function registerAsCall(name, methodName) {
-      self.symbols.registerSchema(name, 'Any', undefined, {
+      self.symbols.registerSchema(name, ast.typeType('Any'), undefined, {
         validate: ast.method(['this'], ast.call(ast.reference(ast.cast(thisVar, 'Any'),
                                                               methodName)))
       });
     }
 
-    this.symbols.registerSchema('Any', 'Any', undefined, {
+    this.symbols.registerSchema('Any', ast.typeType('Any'), undefined, {
```

Comment:
In this case - we're registering a type in a symbol table (by name) - so it can be referenced by other type and path statements.  The "derivedFrom" has migrated to a TypeExpression - something that does not have to be a registered type, but can be one of UnionType, GenericType.